containers/Collateral.ts

```diff
@@ -51,7 +51,11 @@ function useCollateral() {
   const setMaxAllowance = async () => {
     if (contract && empAddress) {
       try {
-        await contract.approve(empAddress, ethers.constants.MaxUint256);
+        const tx = await contract.approve(
+          empAddress,
+          ethers.constants.MaxUint256
+        );
+        notify?.hash(tx.hash);
```

Comment:
I think it sends a [notification](https://docs.blocknative.com/notify)

Reply:
yup. wherever we generate a hash you pass it to the `notify` object and it will track it until it enters a block and provides notifications to the user.

Reply:
What happens if there are multiple transactions pending at the same time for the user?

Reply:
I _think_ it handles this gracefully. I have not tested this specific case.